### PR TITLE
Attempt at the challenge :)

### PR DESCRIPTION
**Pull request by iziamos**
implementation class name: CalculateAverage_iziamos

timing: local time: 37.20user 1.21system 0:01.93elapsed 1986%CPU (0avgtext+0avgdata 3527088maxresident)k

hardware: intel 13700k with 32gb ddr5

Please if you could be kind enough try running with EpsilonGC as well. I haven't defaulted to that because its output appears to affect the test.

### CODE REVIEW

**Comment by iziamos** [comment]
oh, no graal because i dont know how to do that

File: src/main/java/dev/morling/onebrc/CalculateAverage_iziamos.java [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+        }
+
+        return result;
+    }
+
+    private static void processEvent(final ByteBuffer buffer, final PartialResult map) {
+        final long hash = parseId(buffer);
+        final int value = readValue(buffer);
+
+        map.recordResult(hash, value);
```
Comment (gunnarmorling) [inline src/main/java/dev/morling/onebrc/CalculateAverage_iziamos.java:143]: How do you deal with hash collisions here?

**Comment by gunnarmorling** [comment]
One question inline. And could you squash everything into one commit, please? Thx!

File: src/main/java/dev/morling/onebrc/CalculateAverage_iziamos.java (hunk 1 above)
Comment (iziamos) [reply src/main/java/dev/morling/onebrc/CalculateAverage_iziamos.java:143]: I deliberately ignore that. I'm guessing we need something in the rules maybe. I'll make changes

**Pull request closed without merging**